nuxeo-ai-core/src/main/java/org/nuxeo/ai/functions/SaveEnrichmentFunction.java

```diff
@@ -41,7 +42,11 @@ public void accept(EnrichmentMetadata metadata) {
                     DocMetadataService docMetadataService = Framework.getService(DocMetadataService.class);
                     DocumentModel doc = docMetadataService.saveEnrichment(session, metadata);
                     if (doc != null) {
-                        session.saveDocument(doc);
+                        try {
+                            session.saveDocument(doc);
```

Comment:
why not rethrowing the exception? am I missing something, why swallow?

Reply:
Each call is atomic, I came across this error now that autofill properties are being generated.  If the error is thrown up the stack then generic streams re-try functionality kicks in, which is not what we want.

Reply:
ok